### PR TITLE
kvrpcpb: record rollback reason in the CheckTxnStatus response

### PR DESCRIPTION
**Pull request by tiancaiamao**
When the CheckTxnStatus returns rollbacked, provide the rollback reason.

We need this information to debug https://github.com/pingcap/tidb/issues/13397

@coocood @youjiali1995  @MyonKeminta

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -293,6 +293,12 @@ message CheckTxnStatusRequest {
     bool rollback_if_not_exist = 6;
 }
 
+enum RollbackReason {
+    NotByMe = 0;
```
Comment (MyonKeminta) [inline proto/kvrpcpb.proto]: How about `AlreadyRolledBack`, since it also may because duplicated requests

Comment (tiancaiamao) [reply proto/kvrpcpb.proto]: I think it's fine because "Me" can also refer to "this request".

B.T.W, RolledBack or Rollbacked, how to spell

Comment (coocood) [reply proto/kvrpcpb.proto]: `NotByMe` does not look like a reason.

Comment (tiancaiamao) [reply proto/kvrpcpb.proto]: Or we use 'Default' to mean no information?
When the txn status is not rollbacked, we do not need a reason, but we still have to provide a value. @coocood

Comment (coocood) [reply proto/kvrpcpb.proto]: Or `NoReason` as default

**Comment by coocood** [comment]
LGTM

File: proto/kvrpcpb.proto (hunk 1 above)
Comment (tiancaiamao) [reply proto/kvrpcpb.proto]: Done

**Pull request merged**